Libraries/LibC/pwd.cpp

```diff
@@ -125,4 +125,38 @@ struct passwd* getpwent()
     strncpy(__pwdb_entry->shell_buffer, e_shell.characters(), PWDB_STR_MAX_LEN);
     return __pwdb_entry;
 }
+
+int putpwent(const struct passwd* p, FILE* stream)
+{
+    // glibc does some more advanced checks on p's members. We should too eventually.
+    // For now, checking for NULL and setting EINVAL should be good enough :)
+    if (!p || !stream || !p->pw_name || !p->pw_dir || !p->pw_gecos || !p->pw_shell) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (AK::String(p->pw_name).contains(":") || AK::String(p->pw_name).contains("\n")) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (AK::String(p->pw_dir).contains(":") || AK::String(p->pw_dir).contains("\n")) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (AK::String(p->pw_gecos).contains(":") || AK::String(p->pw_gecos).contains("\n")) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    if (AK::String(p->pw_shell).contains(":") || AK::String(p->pw_shell).contains("\n")) {
+        errno = EINVAL;
+        return -1;
+    }
```

Comment:
Allocating all of these into `String`s (twice!) just to check for some characters seems excessive. I'd just use good old `strchr()`